### DISCUSSION TITLE
Using twarc as part of a GitHub Workflow

### DISCUSSION THREAD

**Discussion by DanielBrett** [DocNow/twarc#663] [Q&A] [answered]
Hi all,

I'm looking to put together a simple workflow on GitHub that runs twarc.  The idea is to have it fire weekly, do a twarc search gathering the last 7days of tweets on a topic (or topics), and append the new data onto an ever growing file.  For the most part this all seems doable but my current issue is that I can't figure out how to set up a config file solely using a script with no user interaction.  I get the "Hi I don't see a configuration file yet, so let's make one." message and then the workflow run ends because user interaction was requested.

My question is, can you set up all the config details using commands and no interactions?

Other options to do the same thing would also be appreciated.

Thanks,

Daniel

**Comment by igorbrigadir** [comment, marked as answer]
The best way to do it, would be to create a new secret variable on the git repo settings: `https://github.com/you/your-repo/settings/secrets/actions` called `BEARER_TOKEN` with your bearer token,

And then in the action, run:

```
twarc2 --bearer-token "${{ secrets.BEARER_TOKEN }}" search "dog" >> results.jsonl
```

Or some other combination of `--consumer-key` `--consumer-secret` or whatever - see `twarc2 --help` for the authentication options.

Note the `>>` output - this will append to an existing file, if you leave it out, it will keep overwriting it.

**Reply by DanielBrett** [reply]
That works perfectly!  Thank you so much for the quick response.  I didn't realize that you could do a run with just a bearer token and not the rest of the credentials.
